### DISCUSSION TITLE
Authenticating or auto-login to Github Container Registry? (ghcr.io)

### DISCUSSION THREAD

**Discussion by ringods** [goreleaser/goreleaser#2046] [Q&A] [answered]
Hello,

Here is my project using goreleaser integrated with Github Actions:

https://github.com/ringods/pulumi-resource

When I set a SemVer tag, the release kicks off and it should publish the Docker image to the Github Container Registry. However, this fails:

https://github.com/ringods/pulumi-resource/runs/1826323978

The release process uses a personal access token with `read:packages`, `write:packages` and `delete:packages` scope. I  find references in the code (#1431) that goreleaser should auto-login when `GITHUB_TOKEN` is set. But still, pushing the image fails.

What am I missing here? Should I still login to ghcr.io explicitely?

Greetings,

Ringo

**Comment by caarlos0** [comment, marked as answer]
> I find references in the code (#1431) that goreleaser should auto-login when `GITHUB_TOKEN` is set.

Hi, that is just the docker image, however the github action does not do that.

You'll need to login yourself. [Example](https://carlosbecker.com/posts/multi-platform-docker-images-goreleaser-gh-actions/)

**Reply by ringods** [reply]
Thanks. This worked.
